sapientml_core/templates/model_templates/model.py.jinja

```diff
@@ -22,21 +22,34 @@ model = {{ model_name }}({{ silent }}random_state=random_state_model, {{ params
 from sklearn.multioutput import MultiOutputRegressor
 
 model = MultiOutputRegressor(model)
+
 {% elif is_multioutput_classification %}
 from sklearn.multioutput import MultiOutputClassifier
 
 model = MultiOutputClassifier(model)
 {% endif %}
 {% set xgbclassifier = "XGBClassifier" %}
-{% if model_name == xgbclassifier %}
```

Comment:
As you mentioned, the target is not (0,1,2,...) and the model is XGBClassifier, an error is raised. 
So, we want to use LabelEncoder when
- the targets are categorical (you've implemented in this PR),

or,
 
- the model is XGBClassifier.